semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/DenotationOps.scala

```diff
@@ -118,7 +120,9 @@ trait DenotationOps { self: DatabaseOps =>
 
     private def newInfo: (Option[s.Type], List[g.Symbol]) = {
       val ginfo = {
-        if (gsym.isGetter && gsym.isLazy && !gsym.isClass) {
+        if (gsym.isJavaClass) {
+          gsym.companionClass.info
+        } else if (gsym.isGetter && gsym.isLazy && !gsym.isClass) {
```

Comment:
Looks like the three `isJavaClass` checks above can be replaced by a single condition in `val gsym` above. I'll experiment with this myself after merging this PR.

Reply:
Nice job cleaning this up in #1416